### ISSUE TITLE
There's way too much test code

### ISSUE THREAD

**Issue by C7C8** [Dreadbot/Recycle-Rush#6]
I'll skip the intro and get to the point - there's way too much test code, especially for stuff that was changed that didn't need to be changed. For example, there's an entirely new set of vision routines that have to be tested, although from what I've seen there wasn't really a strong need for them. It also caused a near-rewrite of the vision switching code we already have, which might cause some issues (the NI vision code is extraordinarily picky).

Here's a small list of totally and completely untested stuff:
*The new test mode stuff
*New PID control stuff
*Vision during a match (not new, but was dramatically changed in a way that I really doubt works)
*A brand new, completely untested vision routine that corrects for left/right translation during autonomous. This also dramatically overcomplicated the relatively simple task of driving straight during autonomous, which we weren't really having a huge issue with.
*A brand new batch of drivebase control stuff that didn't need to be changed - the previous control setup worked perfectly, and allowed for easy changing of deadzones (which were removed for some reason) and control settings, which is one of the reasons why we had the previous system in place.

There's a huge amount of stuff here that was changed that needs a lot of testing, and a lot of it didn't even need to be changed. My main concern is everything vision - the new vision refactoring and the new vision code. It might work out, it might not - I honestly don't care, because it's not worth putting in the time to test it at the moment. We have limited time, so can we please focus on fixing/improving what we have instead of rewriting it? These are major, stability-affecting changes that are NOT a top priority now. 

For now, I'm merging (fast-forwarding) the 3TA_Final branch into develop and merging in only certain bugfixes/changes from the Regionals branch (???) to develop, but only stuff like the fix for overlapping DIO pin allocations (or however that commit message was phrased) and possibly the charge compressor during test mode thing (just not the accompanying test code). 

TL;DR We have too much test code that we neither need at the moment nor have the time to test. Let's focus on fixing/improving on the working code that we have rather than rewriting and adding completely new code/changes.
